### DISCUSSION TITLE
Validate vs dry-run: What are the differences and when to use them?

### DISCUSSION THREAD

**Discussion by jtryan0601** [forcedotcom/cli#2484] [Q&A] [answered]
I'm curious about these commands. What are the differences and when to use which? Also, which one has better performance?
a) sf project deploy start --dry-run
b) sf project deploy validate

**Comment by mshanemc** [comment, marked as answer]
`sf project deploy validate` is meant as a precursor to `sf project deploy quick`.  You're wanting to a quick-deploy to production and you want a deployId that will accomplish that.

`sf project deploy start --dry-run` is meant for all other scenarios and is much more flexible.
